astroquery/query.py

```diff
@@ -105,17 +109,27 @@ def request_file(self, cache_location):
         fn = os.path.join(cache_location, self.hash() + ".pickle")
         return fn
 
-    def from_cache(self, cache_location):
+    def from_cache(self, cache_location, cache_timeout):
         request_file = self.request_file(cache_location)
         try:
-            with open(request_file, "rb") as f:
-                response = pickle.load(f)
-            if not isinstance(response, requests.Response):
+            if cache_timeout is None:
+                expired = False
```

Comment:
I don't think this really gains us anything, since we still have to do a specific check for a non-expiring cache.

Reply:
Many other configuration items in `astroquery` use a non-positive value instead of `None` to remove limits, e.g. https://github.com/astropy/astroquery/blob/436245056c3e624b08abb0095a1bbec642ed1526/astroquery/gaia/__init__.py#L32-L34 https://github.com/astropy/astroquery/blob/436245056c3e624b08abb0095a1bbec642ed1526/astroquery/simbad/__init__.py#L26-L29
My suggestion would not simplify code, but it would make it more consistent with existing usage.